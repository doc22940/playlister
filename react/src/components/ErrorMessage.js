import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class ErrorMessage extends React.Component {
  render() {
    const { message } = this.props;
    const classes = classnames({
      'display-none': !message
    });

    return (
      <div id='notfound-wrapper' className={classes}>
        <div className='alert alert-danger'>
          {message}
        </div>
        <hr />
      </div>
    );
  }
}

ErrorMessage.propTypes= {
  message: PropTypes.string.isRequired
};

export default ErrorMessage;

