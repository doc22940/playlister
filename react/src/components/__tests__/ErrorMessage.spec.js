import React from 'react';
import renderer from 'react-test-renderer';

import ErrorMessage from '../ErrorMessage';

it('ErrorMessage renders correctly if no message is provided', () => {
  const tree = renderer
    .create(<ErrorMessage message='' />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('ErrorMessage renders correctly if an error message is provided', () => {
  const tree = renderer
    .create(<ErrorMessage message='Something crazy happened!' />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

