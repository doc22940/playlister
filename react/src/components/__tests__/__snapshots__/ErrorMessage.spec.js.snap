// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly if an error message is provided 1`] = `
<div
  className=""
  id="notfound-wrapper"
>
  <div
    className="jumbotron justify-content-middle text-center"
  >
    <p
      className="lead h3"
    >
      We're trying to find what you are looking for... but it is proving difficult...
    </p>
    <img
      alt="Fat Cat"
      className="img-fluid"
      src="cat.gif"
      style={
        Object {
          "width": "80%",
        }
      }
    />
  </div>
  <br />
</div>
`;

exports[`ErrorMessage renders correctly if no message is provided 1`] = `
<div
  className="display-none"
  id="notfound-wrapper"
>
  <div
    className="jumbotron justify-content-middle text-center"
  >
    <p
      className="lead h3"
    >
      We're trying to find what you are looking for... but it is proving difficult...
    </p>
    <img
      alt="Fat Cat"
      className="img-fluid"
      src="cat.gif"
      style={
        Object {
          "width": "80%",
        }
      }
    />
  </div>
  <br />
</div>
`;
